### PR TITLE
Tweaks and a second README example

### PR DESCRIPTION
**Pull request by timholy**
I learned about this package from @rdeits' exciting JuliaCon talk and have been meaning to explore it ever since. Thanks for creating and releasing it.

This adds a couple of tweaks that slightly smooth the experience on 0.7/1.0 and which support a new README example.

### CODE REVIEW

**Comment by timholy** [comment]
The test seems to fail on 0.6 due to this:
```julia
julia> resid = @expression X'*g - p
ERROR: DimensionMismatch("")
Stacktrace:
 [1] matvecmul!(::Array{Parametron.Functions.AffineFunction{Float64},1}, ::Array{Float64,2}, ::Array{Parametron.Functions.Variable,1}) at /home/tim/.julia/v0.6/Parametron/src/functions.jl:638
 [2] macro expansion at /home/tim/.julia/v0.6/Parametron/src/lazyexpression.jl:59 [inlined]
 [3] evalexpr at /home/tim/.julia/v0.6/Parametron/src/lazyexpression.jl:56 [inlined]
 [4] LazyExpression at /home/tim/.julia/v0.6/Parametron/src/lazyexpression.jl:53 [inlined]
 [5] wrap(::Parametron.LazyExpression{Base.#Ac_mul_B,Tuple{Parametron.Parameter{Array{Float64,2},Parametron.##3#4{Array{Float64,2}},false},Array{Parametron.Functions.Variable,1}}}) at /home/tim/.julia/v0.6/Parametron/src/lazyexpression.jl:85
 [6] optimize_toplevel(::Parametron.LazyExpression{Base.#Ac_mul_B,Tuple{Parametron.Parameter{Array{Float64,2},Parametron.##3#4{Array{Float64,2}},false},Array{Parametron.Functions.Variable,1}}}) at /home/tim/.julia/v0.6/Parametron/src/lazyexpression.jl:193
```
which seems independent of this PR. Let me know if you want it fixed on 0.6, or wish to make the test run only on 0.7+, or want to drop 0.6 altogether.

File: src/parameter.jl [hunk 1]
```diff
@@ -61,6 +61,13 @@ struct Parameter{T, F, InPlace}
 end
 Parameter(f, model) = Parameter{typeof(f())}(f, model)
 
+"""
+    Parameter(a::AbstractArray, model)
+
+Create a new 'in-place' `Parameter` from `a`, whose values will be used to solve `model`.
+"""
+Parameter(a::AbstractArray, model) = Parameter(()->a, model)
```
Comment (tkoolen) [inline src/parameter.jl]: See also https://github.com/tkoolen/Parametron.jl/issues/67 (which needs resolution). I don't think it's a good idea to make `AbstractArray` special in this sense, and the `Parameter{T}(f::F, model) where {T, F}` constructor already takes over the general case. What do you think about the options discussed in #67?

File: src/model.jl [hunk 2]
```diff
@@ -265,3 +265,8 @@ macro objective(model, sense, expr)
         Parametron.setobjective!($model, $sense, Parametron.@expression $expr)
     end |> esc
 end
+
+@static if VERSION >= v"0.7-"
+    # Support constructs like value.(model, x)
+    Broadcast.broadcastable(m::Model) = Ref(m)
```
Comment (tkoolen) [inline src/model.jl:271]: Sure, we can do this. I'm still annoyed that you have to do this for all non-iterable types now.

File: test/model.jl [hunk 3]
```diff
@@ -355,4 +355,27 @@ end
     @test value(model, x) ≈ 1.0 atol=1e-6
 end
 
+@testset "Parameter(A, model)" begin
```
Comment (tkoolen) [inline test/model.jl:358]: Should seed the RNG here (or ideally use a separate `MersenneTwister` with a fixed seed) to avoid random test failures.

**Review by tkoolen — CHANGES_REQUESTED** [review]
Thank you, looks good in general. Please see the inline comments. I'll take a look at the 0.6 failure.

File: src/model.jl (hunk 2 above)
Comment (tkoolen) [reply src/model.jl:271]: To clarify, no action needed, just wanted to vent 🙂.

File: src/parameter.jl (hunk 1 above)
Comment (timholy) [reply src/parameter.jl]: I'll comment there.

**Comment by timholy** [comment]
Since #67 is open and needs resolution, I just decided to strip out anything controversial here. Should be good to go (modulo the 0.6 test failure, which I'm leaving up to you).

**Pull request merged**